data/blog/2022-04-08.robur.md

```diff
@@ -0,0 +1,44 @@
+---
+
+At [Robur](https://robur.coop), we have created many unikernels and services over the years using MirageOS, like [OpenVPN](https://builds.robur.coop/job/openvpn-router/), [CalDAV](https://builds.robur.coop/job/caldav/), a [*Let’s Encrypt*](https://letsencrypt.org/) [solver using DNS](https://builds.robur.coop/job/dns-letsencrypt/), [DNS Resolver](https://builds.robur.coop/job/dns-resolver/), authoritative [DNS Servers storing in a git remote](https://builds.robur.coop/job/dns-primary-git/), and other [reproducible binaries for deployment](https://builds.robur.coop/). We chose OCaml because of its advanced security, compiler speed, and automated memory management. Read more about how Robur benefits from OCaml and MirageOS [on our website](https://robur.coop/Our%20Work/Technology-Employed#OCaml).
+
+Most recently, we worked on [producing and deploying binary unikernels](https://mirage.io/blog/deploying-mirageos-robur), funded by the European Union under the [NGI pointer](https://pointer.ngi.eu) programme.
+
+### OpenVPN
+
+Robur started the development of OpenVPN, a virtual private network protocol, in C rather than OCaml. This choice showed its weaknesses, so we revised it multiple times with an extension to address these issues; however, it soon became clear we needed to start from the beginning and develop it in OCaml using existing libraries and parsers. OCaml improved the project by enhancing security and minimising our codebase.
+
+We created OpenVPN as a MirageOS unikernel to forward all traffic to a single IP address or local network NAT through the OpenVPN tunnel. To increase security even further, we designed a fail-safe that dropped all packages (rather than sending them unencrypted) when the OpenVPN tunnel was down.
+
+This project was funded by [Prototypefund](https://prototypefund.de) in 2019. The [code is available](https://github.com/roburio/openvpn).
+
+### CalDAV
+
+Robur started the CalDAV project back in 2017 with a grant from the [Prototype Fund](https://prototypefund.de), which came with the stipulation that it had to be created within six months! Thus the first version of CalDAV emerged from those efforts. Afterward, [Tarides](https://tarides.com) sponsored Robur so we could continue CalDAV’s development and draft an inital CardDAV implementation, resulting in the version now available.
+
+As you might’ve guessed from the name, CalDAV is a protocol to synchronise calendars, and it works on a robust server that’s relatively easy to configure and maintain. This enables more people to run our own digital infrastructure rather than rely on an outside source. Robur’s CalDAV provides considerable security, due to its minimal codebase, and stores its data on Git for version control. We even set up a live *test server* at [calendar.robur.coop](calendar.robur.coop) that contains the [calDavZAP user interface](https://inf-it.com/open-source/clients/caldavzap/), accessible with any username and password.
```

Comment:
```suggestion
As you might’ve guessed from the name, CalDAV is a protocol to synchronise calendars, and it works on a robust server that’s relatively easy to configure and maintain. This enables more people to run our own digital infrastructure rather than rely on an outside source. Robur’s CalDAV provides considerable security, due to its minimal codebase, and stores its data on Git for version control. We even set up a live *test server* at [calendar.robur.coop](calendar.robur.coop) that contains the [CalDavZAP user interface](https://inf-it.com/open-source/clients/caldavzap/), accessible with any username and password.
```